sklearn/utils/estimator_checks.py

```diff
@@ -360,8 +360,17 @@ def _generate_class_checks(Estimator):
 
 def _mark_xfail_checks(estimator, check, pytest):
     """Mark estimator check pairs with xfail"""
+    if isinstance(estimator, type):
+        # try to construct estimator to get tags, if it is unable to then
+        # return the estimator class
+        try:
+            xfail_checks = _safe_tags(_construct_instance(estimator),
```

Comment:
`_construct_instance` is potentially costly, but I also don't see a way around it.

Reply:
I'm not concerned given our conventions